languages/go/exercises/concept/basic-slices/.meta/example.go

```diff
@@ -0,0 +1,69 @@
+package _meta
+
+// GetItem retrieves an item from a slice at given position. The second return value indicates whether
+// a the given index existed in the slice or not.
+func GetItem(slice []uint8, index int) (uint8, bool) {
+	if len(slice) <= index || index < 0 {
+		return 0, false
+	}
+	return slice[index], true
+}
+
+// SetItem writes an item to a slice at given position overwriting an existing value.
+// If the index is out of range it is be appended.
+func SetItem(slice []uint8, index int, value uint8) []uint8 {
+	if len(slice) <= index || index < 0 {
+		return append(slice, value)
+	}
+	slice[index] = value
+	return slice
+}
+
+// PrefilledSlice creates a slice of given length and prefills it with the given value.
+func PrefilledSlice(value, length int) []int {
+	if length < 1 {
+		return nil
+	}
+
+	var s = make([]int, 0, length)
+	for i := 0; i < length; i++ {
+		s = append(s, value)
+	}
+	return s
+}
+
+// NumberRow fills a slice with the numbers 1, 2, 3, etc. It returns as many numbers as are necessary for the sum
+// of the slice to be equal or larger than `sumMin`.
+func NumberRow(sumMin int) []int {
+	var (
+		sum int
+		s   []int
+	)
+	for i := 1; true; i++ {
+		if sumMin <= sum {
+			break
+		}
+		s = append(s, i)
+		sum += i
+	}
+	return s
+}
+
+// RemoveItemPure removes an item without changing the input values or other side effects.
+// In functional programming this is called a `pure` function.
+func RemoveItemPure(slice []int, index int) []int {
```

Comment:
I don't see immutability listed in issue #530. Is the goal here to teach about immutability, or should people already be familiar with it?

Reply:
@ErikSchierboom: Go doesn't really have immutability. There is the `const` keyword which basically works for strings and numbers, but not for `slices`, `maps`, etc. `string` itself is immutable but you can reassign the variable (if not declared with `const`), which we'll cover in some exercise. But that's it.

Any ideas? Maybe talk about how changing a value in the slice affects all uses of that slice? What would you call that concept?

Reply:
Hmmm, I don't know. Maybe there is no such Concept. I assumed there was immutability, but that is apparently not a thing.

Reply:
@ErikSchierboom We could make that a concept... but then I wouldn't know how it applies to this issue. Unless "not touching" but "copying" the slice is part of the immutability concept. Is it?